code/game/objects/items.dm

```diff
@@ -165,7 +166,11 @@
 	else
 		if(isliving(src.loc))
 			return
-	user.put_in_active_hand(src)
+	if(user.put_in_active_hand(src))
+		if (isturf(old_loc))
```

Comment:
A thought here.
If the old loc itself is on a turf, then you could also do the animation.
For example, when taking things out of a backpack/box/toolbox which is on a table

change this to if (isturf(old_loc) || isturf(old_loc.loc))